test/metabase/test/redefs.clj

```diff
@@ -5,22 +5,28 @@
    [mb.hawk.parallel]
    [metabase.plugins.classloader :as classloader]
    [methodical.core :as methodical]
+   [toucan2.connection :as t2.connection]
    [toucan2.tools.with-temp :as t2.with-temp]))
 
+(def ^{:dynamic true
+       :private true}
+  *in-with-temp* false)
+
 (methodical/defmethod t2.with-temp/do-with-temp* :around :default
-  "Initialize the DB before doing the other with-temp stuff. Make sure metabase.test.util is loaded. Make sure we are
-  not inside a parallel test."
+  "Initialize the DB before doing the other with-temp stuff. Make sure metabase.test.util is loaded.
+  Make sure the the "
   [model attributes f]
   (classloader/require 'metabase.test.initialize)
   ((resolve 'metabase.test.initialize/initialize-if-needed!) :db)
   ;; so with-temp-defaults are loaded
   (classloader/require 'metabase.test.util)
-  ;; disallow with-temp inside parallel tests.
-  ;;
-  ;; TODO -- there's not really a reason that we can't use with-temp in parallel tests -- it depends on the test -- so
-  ;; once we're a little more comfortable with the current parallel stuff we should remove this restriction.
-  (mb.hawk.parallel/assert-test-is-not-parallel "with-temp")
```

Comment:
You'll want to update `.clj-kondo/hooks/clojure/test.clj` as well to remove the different versions of `with-temp` from the list of things to warn about inside `^:parallel` tests

Reply:
removed it here https://github.com/metabase/metabase/pull/33738/commits/610795c02d898ef0a936794ed46fad84dd612e05